test/integration/targets/pip/tasks/pip.yml

```diff
@@ -256,3 +256,28 @@
   assert:
     that:
       - not pip_install_empty.changed
+
+# https://github.com/ansible/ansible/issues/37912
+- name: install pip throught pip into fresh virtualenv for test pip without virtualenv
+  pip:
+    name: pip
+    virtualenv: "{{ output_dir }}/pipenv"
+  register: pip_install_venv4test
+
+- name: test for previos task not fail
+  assert:
+    that:
+      - "pip_install_venv4test.state == 'present'"
+
+- name: pip chdir without virtualenv
+  pip:
+    chdir: "{{ output_dir }}/"
+    executable: "{{ output_dir }}/pipenv/bin/pip"
```

Comment:
This somehow contradicts, that you concatenate `chdir` with `executable` in the module.
In the end, i think the documentation of `chdir` is rather poor. It leaves too much room for interpretation. And the only thing it states to do (changing the PWD) is never actually done. Maybe we should simply remove it.

Reply:
If you do not set an executable and virtualenv it changes the directory just fine and is useful, see #40793 
IMHO the uncontroversal change to reference the env variable only when it is set should be shipped already. The discussion how to handle the other issues can be done separately.